src/lib.rs

```diff
@@ -782,7 +784,10 @@ impl<Types: IpfsTypes> Ipfs<Types> {
         }
     }
 
-    /// Establishes the local node as a provider of a value for the given Cid.
+    /// Establishes the node as a provider of a block with the given Cid: it publishes a provider
+    /// record with the given key (Cid) and the node's PeerId to the peers closest to the key. The
+    /// publication of provider records is periodically repeated as per the interval specified in
+    /// `libp2p`'s  `KademliaConfig`.
     pub async fn provide(&self, cid: Cid) -> Result<(), Error> {
```

Comment:
I wonder if this should include a warning on "provides are not persisted across launches". Probably not. I'd like to implement it in the near future.

Reply:
We may need to want to create an issue tracking of all things persistent. Is there /remind functionality on github? :) I'll add a note for myself.